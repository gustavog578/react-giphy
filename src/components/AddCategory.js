import React, { useState } from 'react'
import { PropTypes } from "prop-types";

export const AddCategory = ({setCategories}) => {
// poner '' para que el state inicial no sea undefined
const [Inputvalue, setInputvalue] = useState('')



const handleInputChange = (e) => {
    setInputvalue(e.target.value);
}


const handleSubmit = (e) => {
    e.preventDefault();
    
    if(Inputvalue.trim().length > 2){
        setCategories(cats => [Inputvalue, ...cats])
        setInputvalue('')
    }

}


return (
   
        <form onSubmit={ handleSubmit }>         
            <input
                type="text"
                value={ Inputvalue }
                onChange={ handleInputChange }
            />
        </form>


    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}