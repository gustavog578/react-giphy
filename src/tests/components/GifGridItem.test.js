import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';

// Debe estar aqui debajo
import { shallow } from "enzyme"

describe('Test GifGridItem', () => {
    
    const title = 'Un titulo';
    const url='http://localhost:3030';
    const wrapper = shallow( <GifGridItem title= {title} url={url}  /> )


    test('debe momstrar el componente correctamente', () => {
      
        expect(wrapper).toMatchSnapshot()
    })

    test('Debe tener un parrafo con el title', () =>{

        const p = wrapper.find('p');
        expect( p.text().trim()).toBe( title );
    })
    
    test('Debe tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img');
        expect( img.prop('src')).toBe(url);
        expect( img.prop('alt')).toBe(title);

    })

    test('Debe tener animate_fadeIn', () => {

        const div = wrapper.find('div');
        // dos maneras de preguntar
        expect( div.prop('className')).toContain('animate__fadeIn')
        expect( div.prop('className').includes('animate__fadeIn')).toBe(true)

    })


})

