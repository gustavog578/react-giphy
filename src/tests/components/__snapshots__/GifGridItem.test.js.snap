// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifGridItem debe momstrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="http://localhost:3030"
  />
  <p>
    Un titulo
  </p>
</div>
`;
